# Data Analysis on Haberman Cancer Survival dataset

This dataset can be obtained from Kaggle: https://www.kaggle.com/gilsousa/habermans-survival-data-set

In [1]:
import numpy as numpy
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as pyplot

haberman = pd.read_csv("haberman.csv")
haberman.head()

age  year  nodes  status
0   30    64      1       1
1   30    62      3       1
2   30    65      0       1
3   31    59      2       1
4   31    65      4       1

In [4]:
print(haberman.shape)

(306, 4)


There is 306 datapoints with 4 features.

In [5]:
# Print the column (feature) names
print(haberman.columns)

Index(['age', 'year', 'nodes', 'status'], dtype='object')


Feature information:

- Age: Represents the age of the patient

- Year: The year the patient had the operation

- Nodes: The number of Axillary (Lymph) nodes

- Status: Indicates how long the patient survived. (1 = More than 5 years, 2 = Within 5 years)


In [7]:
haberman["status"].value_counts()

1    225
2     81
Name: status, dtype: int64

Based on the above figures, we can see that out of the 306 patients, 225 patients survived for more than 5 years and 81 patients died within 5 years

In [6]:
haberman.describe()

age        year       nodes      status
count  306.000000  306.000000  306.000000  306.000000
mean    52.457516   62.852941    4.026144    1.264706
std     10.803452    3.249405    7.189654    0.441899
min     30.000000   58.000000    0.000000    1.000000
25%     44.000000   60.000000    0.000000    1.000000
50%     52.000000   63.000000    1.000000    1.000000
75%     60.750000   65.750000    4.000000    2.000000
max     83.000000   69.000000   52.000000    2.000000

The above provides statistical information based on each attribute